components/patient-access-rules/rest/src/main/java/org/phenotips/data/permissions/rest/internal/DefaultDomainObjectFactory.java

```diff
@@ -0,0 +1,233 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.permissions.rest.internal;
+
+import org.phenotips.data.Patient;
+import org.phenotips.data.permissions.Collaborator;
+import org.phenotips.data.permissions.Owner;
+import org.phenotips.data.permissions.PatientAccess;
+import org.phenotips.data.permissions.PermissionsManager;
+import org.phenotips.data.permissions.Visibility;
+import org.phenotips.data.permissions.rest.CollaboratorResource;
+import org.phenotips.data.permissions.rest.DomainObjectFactory;
+import org.phenotips.data.permissions.rest.Relations;
+import org.phenotips.data.permissions.script.SecurePatientAccess;
+import org.phenotips.data.rest.model.CollaboratorRepresentation;
+import org.phenotips.data.rest.model.CollaboratorsRepresentation;
+import org.phenotips.data.rest.model.Link;
+import org.phenotips.data.rest.model.PatientVisibilityRepresentation;
+import org.phenotips.data.rest.model.UserSummary;
+
+import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.stability.Unstable;
+
+import java.util.Collection;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.core.UriInfo;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Default implementation of {@link DomainObjectFactory}.
+ *
+ * @version $Id$
+ * @since 1.2M5
+ */
+@Unstable
+@Component
+@Singleton
+public class DefaultDomainObjectFactory implements DomainObjectFactory, Initializable
+{
+    /** Provides access to the underlying data storage. */
+    @Inject
+    private DocumentAccessBridge documentAccessBridge;
+
+    /** Parses string representations of document references into proper references. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<EntityReference> referenceResolver;
+
+    @Inject
+    private PermissionsManager manager;
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private EntityReferenceSerializer<String> entitySerializer;
+
+    private EntityReference userObjectReference;
+
+    private EntityReference groupObjectReference;
```

Comment:
`userObjectReference` and `groupObjectReference` are used only in the inner class `NameEmail`. It might be tidier to move the class, the method `initialize()` and the two members to a separate file.